### PR TITLE
use session endpoint metadata for oauth logout

### LINKED ISSUE #22020: Auth0 Logout not working properly when using a custom domain at auth0
##### **Overview of the issue**
Hi,

in our set-up we are using auth0 as identity provider, where auth0 is having a custom host name configured. The default LogoutResource detects auth0 by checking the issuerUri, see here in the imperative sample:

https://github.com/jhipster/generator-jhipster/blob/28005f01dc427f62c061dcab5cd6f62e5063377a/generators/server/templates/src/main/java/package/web/rest/LogoutResource_imperative.java.ejs#L56-L60

Since we are having a custom host name (enterprise...) where auth0.com is not mentioned at all, we fall back into the else branch, where there is the call `this.registration.getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString()`. 

Since auth0 is seemingly not publishing this as part of the `/.well-known/openid-configuration` endpoint, the `.toString()` produces a `NullPointerException`... in any case auth0 seems to require some special handling when it comes to Logouts, so we would probably require a different detection mechanism for auth0...

```
2023-05-05 17:17:30.148 ERROR   [XNIO-1 task-1] o.z.p.s.c.AdviceTraits - Internal Server Error
java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "java.util.Map.get(Object)" is null
	at de.mt.dataview.web.rest.LogoutResource.logout(LogoutResource.java:41)
	at de.mt.dataview.web.rest.LogoutResource$$FastClassBySpringCGLIB$$c64cca3a.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
```

##### **Motivation for or Use Case**
Auth0 detection not working in any case.

##### **Reproduce the error**
See above 

##### **Suggest a Fix**
Maybe allow to define auth0 as idp in a config when it is not possible to detect it implicitly... any ideas for other detection mechanisms welcome :-)

##### **JHipster Version(s)**
7.9.3

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by atomfrede**
no special handling for auth0 anymore, using well known meta data.

closes #22020

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Review by mraible — APPROVED** [review]

File: generators/server/templates/src/main/java/_package_/web/rest/LogoutResource_reactive.java.ejs [hunk 1]
```diff
@@ -59,19 +59,13 @@ public class LogoutResource {
...
     private Map<String, String> prepareLogoutUri(ServerHttpRequest request, ClientRegistration clientRegistration, OidcIdToken idToken) {
         StringBuilder logoutUrl = new StringBuilder();
-        String issuerUri = clientRegistration.getProviderDetails().getIssuerUri();
-        if (issuerUri.contains("auth0.com")) {
-            logoutUrl.append(issuerUri.endsWith("/") ? issuerUri + "v2/logout" : issuerUri + "/v2/logout");
-        } else {
-            logoutUrl.append(clientRegistration.getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString());
-        }
+
+        logoutUrl.append(this.registration.getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString());
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/_package_/web/rest/LogoutResource_reactive.java.ejs:63]:
```suggestion
        logoutUrl.append(clientRegistration..getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString());
```

File: generators/server/templates/src/main/java/_package_/web/rest/LogoutResource_reactive.java.ejs [hunk 2]
```diff
@@ -59,19 +59,13 @@ public class LogoutResource {
...
     private Map<String, String> prepareLogoutUri(ServerHttpRequest request, ClientRegistration clientRegistration, OidcIdToken idToken) {
         StringBuilder logoutUrl = new StringBuilder();
-        String issuerUri = clientRegistration.getProviderDetails().getIssuerUri();
-        if (issuerUri.contains("auth0.com")) {
-            logoutUrl.append(issuerUri.endsWith("/") ? issuerUri + "v2/logout" : issuerUri + "/v2/logout");
-        } else {
-            logoutUrl.append(clientRegistration.getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString());
-        }
+
+        logoutUrl.append(clientRegistration..getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString());
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/_package_/web/rest/LogoutResource_reactive.java.ejs:63]:
```suggestion
        logoutUrl.append(clientRegistration.getProviderDetails().getConfigurationMetadata().get("end_session_endpoint").toString());
```

**Review by mraible — APPROVED** [review]

**Review by DanielFran — APPROVED** [review]

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/175027

**Comment by DanielFran** [comment]
@atomfrede approved

**Pull request merged**